### PR TITLE
[X86][CFE] Correct parameter type of _cmpccxadd_epi64

### PR DESCRIPTION
**Pull request by FreddyLeaf**
This fixes correctness of https://gcc.godbolt.org/z/vexf5fW5r

### CODE REVIEW

**Comment by FreddyLeaf** [comment]
First, for usages like in https://gcc.godbolt.org/z/vexf5fW5r, clang didn't report warning/errors. Furthermore, it calls the builtin with a 4 byte aligned ptr instead of 8, which brought correctness issues.

File: clang/test/CodeGen/X86/cmpccxadd-builtins-error.c [hunk 1]
```diff
@@ -10,3 +10,7 @@ int test_cmpccxadd32(void *__A, int __B, int __C) {
 long long test_cmpccxadd64(void *__A, long long __B, long long __C) {
   return _cmpccxadd_epi64(__A, __B, __C, 16); // expected-error {{argument value 16 is outside the valid range [0, 15]}}
 }
+
+long long test_cmpccxadd64_2(int *__A, long long __B, long long __C) {
+  return _cmpccxadd_epi64(__A, __B, __C, 3); // expected-error {{incompatible pointer types passing 'int *' to parameter of type 'long long *'}}
```
Comment (phoebewang) [inline clang/test/CodeGen/X86/cmpccxadd-builtins-error.c:15]: Can this be `expected-warning`?

Comment (FreddyLeaf) [reply clang/test/CodeGen/X86/cmpccxadd-builtins-error.c]: yes, [f115369](https://github.com/llvm/llvm-project/pull/114367/commits/f1153692284748c620ef26992b6947140aa690d2)

**Review by phoebewang — APPROVED** [review]
LGTM, thanks!

**Pull request merged**